client/my-sites/post-type-list/index.jsx

```diff
@@ -8,255 +8,214 @@ import PropTypes from 'prop-types';
+		this.scrollListener.cancel(); // Cancel any pending scroll events
 	}
 
-	handleWindowResize = () => {
-		this.setState( {
-			windowWidth: window.innerWidth,
-		} );
-	};
+	estimatePageCountFromPosts( posts ) {
+		// When loading posts from persistent storage, we want to avoid making
+		// a bunch of sequential requests when the user scrolls down to the end
+		// of the list.  However, we want to still request the posts, in case
+		// some data has changed since the last page reload.  This will spawn a
+		// number of concurrent requests for different pages of the posts list.
 
-	getInitialRequestedPages( props ) {
-		// If we have no posts or we're otherwise not expecting any posts to be
-		// rendered, request the first page, since setRequestedPages won't be
-		// called if row count is 0.
-		if ( 0 === size( props.posts ) ) {
-			return [ 1 ];
+		if ( ! posts || ! posts.length ) {
+			return 1;
 		}
 
-		return [];
-	}
+		const query = this.props.query || {};
+		const postsPerPage = query.number || DEFAULT_POST_QUERY.number;
+		const pageCount = Math.ceil( posts.length / postsPerPage );
 
-	getPageForIndex( index ) {
-		const { query, lastPage } = this.props;
-		const perPage = query.number || DEFAULT_POSTS_PER_PAGE;
-		const page = Math.ceil( index / perPage );
+		// Avoid making more than 5 concurrent requests on page load.
+		return Math.min( pageCount, 5 );
+	}
 
-		return Math.max( Math.min( page, lastPage || Infinity ), 1 );
+	getScrollTop() {
+		const { scrollContainer } = this.props;
+		if ( ! scrollContainer ) {
+			return null;
+		}
+		if ( scrollContainer === document.body ) {
+			return 'scrollY' in window
+				? window.scrollY
+				: document.documentElement.scrollTop;
+		}
+		return scrollContainer.scrollTop;
 	}
 
-	setRequestedPages( { startIndex, stopIndex } ) {
-		if ( ! this.props.query ) {
+	maybeLoadNextPage() {
+		const { scrollContainer, lastPage, isRequestingPosts } = this.props;
+		if ( ! scrollContainer ) {
 			return;
 		}
-
-		const { requestedPages } = this.state;
-		const pagesToRequest = difference(
-			range(
-				this.getPageForIndex( startIndex - LOAD_OFFSET ),
-				this.getPageForIndex( stopIndex + LOAD_OFFSET ) + 1
-			),
-			requestedPages
-		);
-
-		if ( ! pagesToRequest.length ) {
+		const scrollTop = this.getScrollTop();
+		const { scrollHeight, clientHeight } = scrollContainer;
+		if (
+			typeof scrollTop !== 'number' ||
+			typeof scrollHeight !== 'number' ||
+			typeof clientHeight !== 'number'
+		) {
 			return;
 		}
-
-		this.setState( {
-			requestedPages: requestedPages.concat( pagesToRequest ),
-		} );
-	}
-
-	isLastPage() {
-		const { lastPage, requestingLastPage } = this.props;
-		const { requestedPages } = this.state;
-
-		return includes( requestedPages, lastPage ) && ! requestingLastPage;
+		const pixelsBelowViewport = scrollHeight - scrollTop - clientHeight;
+		const { maxRequestedPage } = this.state;
+		if (
+			pixelsBelowViewport <= 200 &&
```

Comment:
Could we try 400 here? I tried this myself and it made the whole "waiting for more posts to load" thing feel like less of an issue. Also, should we make this a constant?

Reply:
Let's definitely try that (and make it a constant)! If it doesn't get in this PR I'll create a separate story to tweak this.

Reply:
Done in 33861d5.